### PR TITLE
Feature/add component tests

### PR DESCRIPTION
**Pull request by ChrysmOre**
### What

Added component tests to service.

Uses real Kafka cluster via docker-compose.

### How to review

Check tests cover appropriate range of use cases (and they that they pass!)

### Who can review

Anyone

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -2,26 +2,35 @@ package main
...
+	logFileName = "log-output.txt"
+)
+
+type ComponentTest struct{}
 
 func (f *ComponentTest) InitializeScenario(ctx *godog.ScenarioContext) {
-	component := steps.NewComponent()
+	component, err := steps.NewComponent()
+	if err != nil {
+		log.Panicf("unable to initialise component: %s", err)
```
Comment (cookel2) [inline main_test.go:27]: I think it's best not to panic : ) - that's what I've been told anyway
```suggestion
		os.Exit(1)
```

File: main_test.go [hunk 2]
```diff
@@ -2,26 +2,35 @@ package main
...
+	}
 
 	ctx.BeforeScenario(func(*godog.Scenario) {
-		component.Reset()
+		if err := component.Reset(); err != nil {
+			log.Panicf("unable to initialise scenario: %s", err)
```
Comment (cookel2) [inline main_test.go:32]:
```suggestion
			os.Exit(1)
```

File: ci/component.yml [hunk 3]
```diff
@@ -5,11 +5,20 @@ platform: linux
...
+  path: bash
+  args:
+    - -exc
+    - |
+      source /docker-lib.sh
+      start_docker
+      dp-import-cantabular-dataset/ci/scripts/component.sh
+
+caches:
+  - path: /var/run/docker.sock
```
Comment (cookel2) [inline ci/component.yml:24]: new line needed here

File: main_test.go (hunk 1 above)
Comment (ChrysmOre) [reply main_test.go:27]: There are times when panicking is better than a raw exit.. although here it's basically of no consequence either way :) When calling os.Exit() no deferred functions are called, which can be an issue. Panicking is usually frowned upon when it's done at random points through the code but that at the very top level it's usually fine. I'm happy to change it here to keep in line with code elsewhere though as it really doesn't matter either way

**Review by cookel2 — APPROVED** [review]
it ran successfully in my local env

**Pull request merged**